### PR TITLE
PAYEE-1169 - Using optionT in the cacheConnector

### PR DESCRIPTION
**Pull request by brendanstraughan**
- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/tai/connectors/CacheConnector.scala [hunk 1]
```diff
@@ -85,18 +85,16 @@ class CacheConnector @Inject()(
   def find[T](cacheId: CacheId, key: String = defaultKey)(implicit reads: Reads[T]): Future[Option[T]] =
     if (mongoConfig.mongoEncryptionEnabled) {
       val jsonDecryptor = new JsonDecryptor[T]()
-      taiCacheRepository.findById(cacheId.value) map {
-        case Some(cache) =>
+      OptionT(taiCacheRepository.findById(cacheId.value)).map {
```
Comment (sergi-hmrc) [inline app/uk/gov/hmrc/tai/connectors/CacheConnector.scala:88]: to refactor a bit your code u could extract a common method for taiCache and taiUpdate:

`private def findById[T](cacheId: CacheId, key: String)(f: String => Future[Option[CacheItem]])(implicit reads: Reads[T]): Future[Option[T]] =
    if (mongoConfig.mongoEncryptionEnabled) {
      val jsonDecryptor = new JsonDecryptor[T]()
      OptionT(f(cacheId.value)).map {
        cache =>
          (cache.data \ key).validateOpt[Protected[T]](jsonDecryptor).asOpt.flatten.map(_.decryptedValue)
      }.value.map(_.flatten)
    } recover {
      case JsResultException(_) => None
    } else {
      OptionT(f(cacheId.value)).map {
        cache => (cache.data \ key).validateOpt[T].asOpt.flatten
      }.value.map(_.flatten)
    }

  def find[T](cacheId: CacheId, key: String = defaultKey)(implicit reads: Reads[T]): Future[Option[T]] =
    findById(cacheId, key)(taiCacheRepository.findById)

  def findUpdateIncome[T](cacheId: CacheId, key: String = defaultKey)(implicit reads: Reads[T]): Future[Option[T]] =
    findById(cacheId, key)(taiUpdateIncomeCacheRepository.findById)`

File: app/uk/gov/hmrc/tai/connectors/CacheConnector.scala [hunk 2]
```diff
@@ -111,39 +109,28 @@ class CacheConnector @Inject()(
...
-      }
+      }.value.map(_.flatten)
     }
 
   def findJson(cacheId: CacheId, key: String = defaultKey): Future[Option[JsValue]] =
     find[JsValue](cacheId, key)
 
   def findSeq[T](cacheId: CacheId, key: String = defaultKey)(implicit reads: Reads[T]): Future[Seq[T]] =
-    if (mongoConfig.mongoEncryptionEnabled) {
+
```
Comment (sergi-hmrc) [inline app/uk/gov/hmrc/tai/connectors/CacheConnector.scala:122]: `findOptSeq[T](cacheId, key).map(_.getOrElse(Nil))`

File: app/uk/gov/hmrc/tai/connectors/CacheConnector.scala [hunk 3]
```diff
@@ -23,21 +23,22 @@ import play.api.Configuration
...
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class CacheConnector @Inject()(
-  taiCacheRepository: TaiCacheRepository,
-  taiUpdateIncomeCacheRepository: TaiUpdateIncomeCacheRepository,
-  mongoConfig: MongoConfig,
-  configuration: Configuration)(implicit ec: ExecutionContext)
-    extends MongoFormatter {
+                                taiCacheRepository: TaiCacheRepository,
```
Comment (sergi-hmrc) [inline app/uk/gov/hmrc/tai/connectors/CacheConnector.scala:34]: remove these spaces

**Review by sergi-hmrc — APPROVED** [review]

**Review by Jake-Raffe — APPROVED** [review]

**Review by vivekhmrc — APPROVED** [review]

**Review by PerrettJ4 — APPROVED** [review]
clean

**Pull request merged**